nx/lib/nx.ex

```diff
@@ -11581,6 +11581,14 @@ defmodule Nx do
     end
   end
 
+  defp reshape_with_order(tensor, shape, false), do: Nx.reshape(tensor, shape)
+
+  defp reshape_with_order(tensor, shape, true) do
+    shape = shape |> Tuple.to_list() |> Enum.reverse() |> List.to_tuple()
+
+    Nx.reshape(tensor, shape) |> Nx.transpose()
```

Comment:
Can't we transpose the shape itself instead?

Reply:
I'm a bit confused, though, because the docs for Nx.transpose say:

```
If no axes are given, the default behavior is to reverse the order of the
original tensor's axes.
```

Reply:
```elixir
iex(4)> shape = {2, 3, 4}
{2, 3, 4}
iex(5)> Nx.iota(shape |> Tuple.to_list() |> Enum.reverse() |> List.to_tuple()) |> Nx.transpose()
#Nx.Tensor<
  s64[2][3][4]
  [
    [
      [0, 6, 12, 18],
      [2, 8, 14, 20],
      [4, 10, 16, 22]
    ],
    [
      [1, 7, 13, 19],
      [3, 9, 15, 21],
      [5, 11, 17, 23]
    ]
  ]
>
```

Reply:
Resolving the conversation due to self-rubberducking :)

Reply:
Basically, Fortran order means column-major instead of row-major